### PR TITLE
Allow for GT to be a nocall if GQ and PL[0] are zero instead of homref

### PR DESCRIPTION
**Pull request by nalinigans**
In response to https://github.com/broadinstitute/gatk/pull/8715, allow for GT to be emitted as a nocall for samples that have GQ and PL[0] as zeroes.

@mlathara, for some reason, htsjdk is emitting GT as ./. for haploids in a mixed ploidy position. htslib is emitting the correct haploid nocall for the same. This seems to work correctly from gatk though. Will be looking into this, but any ideas appreciated. The htsjdk version is the same in both gatk and GenomicsDB btw.

### CODE REVIEW

File: src/main/cpp/src/query_operations/broad_combined_gvcf.cc [hunk 1]
```diff
@@ -663,7 +663,16 @@ void BroadCombinedGVCFOperator::handle_FORMAT_fields(const Variant& variant) {
...
   uint64_t num_elements = 0ull;
-  int* int_vec = 0;     //will point to same address as ptr, but as int*
+  int* GT_vec = 0; // will point to same address as ptr, but as int*
+  int* DP_vec = 0; // ditto
+  //Hold values to figure out if any samples have GQ==0 and PL[0]==0 after the loop
+  //see https://github.com/broadinstitute/gatk/pull/8715
+  std::vector<bool> GQ_val(variant.get_num_calls(), false);
+  std::vector<bool> PL_val(variant.get_num_calls(), false);
+  bool some_GQ_vals_zero = false;
+  bool some_PL_first_vals_zero = false;
```
Comment (mlathara) [inline src/main/cpp/src/query_operations/broad_combined_gvcf.cc:673]: nit: we don't seem to use these `some_*` booleans anywhere

**Review by mlathara — APPROVED** [review]
lgtm: only minor comment. Not sure about the mixed ploidy stuff myself -- could you add a TODO for those in the tests?

File: src/main/cpp/src/query_operations/broad_combined_gvcf.cc (hunk 1 above)
Comment (nalinigans) [reply src/main/cpp/src/query_operations/broad_combined_gvcf.cc:673]: I am using sets now (see lines 674/675 below)and I thought I committed those changes. Will remove them, thanks!

**Comment by nalinigans** [comment]
> lgtm: only minor comment. Not sure about the mixed ploidy stuff myself -- could you add a TODO for those in the tests?

Yes, I will add a todo for these. Thanks!

**Pull request merged**